ArduPlane/commands_logic.cpp

```diff
@@ -1184,13 +1184,26 @@ void Plane::nav_script_time_done(uint16_t id)
 // support for NAV_SCRIPTING mission command
 bool Plane::set_target_throttle_rate_rpy(float throttle_pct, float roll_rate_dps, float pitch_rate_dps, float yaw_rate_dps)
 {
-    if (!nav_scripting_active()) {
-        return false;
-    }
+    if (nav_scripting_enabled) {      
     nav_scripting.roll_rate_dps = roll_rate_dps;
     nav_scripting.pitch_rate_dps = pitch_rate_dps;
     nav_scripting.yaw_rate_dps = yaw_rate_dps;
     nav_scripting.throttle_pct = throttle_pct;
+    nav_scripting.current_ms = AP_HAL::millis();
     return true;
+    } else {
+    return false;
+    }
+}
+
+void Plane::nav_scripting_enable(uint8_t mode)
+{
+   if (control_mode->mode_number() != mode) {
```

Comment:
i think we should refuse in Q modes, unless we actually implement the desired rates in Q modes (eg. tailsitters could do backflips)